samples/custom_module/sync/main.c

```diff
@@ -90,13 +90,14 @@ int main(int argc, char** argv) {
   const int32_t input_data[5] = {1, 2, 3, 4, 5};
   const iree_hal_dim_t shape[1] = {IREE_ARRAYSIZE(input_data)};
   iree_hal_buffer_view_t* input_view = NULL;
-  IREE_CHECK_OK(iree_hal_buffer_view_allocate_buffer(
+  IREE_CHECK_OK(iree_hal_buffer_view_allocate_buffer_copy(
+      iree_runtime_session_device(session),
       iree_runtime_session_device_allocator(session), IREE_ARRAYSIZE(shape),
       shape, IREE_HAL_ELEMENT_TYPE_INT_32,
       IREE_HAL_ENCODING_TYPE_DENSE_ROW_MAJOR,
       (iree_hal_buffer_params_t){
           .type = IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL,
-          .access = IREE_HAL_MEMORY_ACCESS_READ,
+          .access = IREE_HAL_MEMORY_ACCESS_ALL,
```

Comment:
Why did these samples change from `ACCESS_READ` to `ACCESS_ALL`?

Reply:
they were relying on the initial_data to write into a read-only buffer - this is a bit of a regression but we could add an iree_hal_buffer_protect that allowed for decreasing access after the fact (maybe, threading is hard) to replace it if we wanted.